core/services/assets/Registry.php

```diff
@@ -135,62 +96,60 @@ public function getI18nRegistry()
         return $this->i18n_registry;
     }
 
+
     /**
      * Callback for the WP script actions.
      * Used to register globally accessible core scripts.
      * Also used to add the eejs.data object to the source for any js having eejs-core as a dependency.
      *
+     * @since $VID:$
+     * @throws InvalidDataTypeException
      */
-    public function scripts()
+    public function registerScriptsAndStyles()
     {
-        global $wp_version;
-        wp_register_script(
-            'ee-manifest',
-            $this->getJsUrl(self::ASSET_NAMESPACE, 'manifest'),
-            array(),
-            null,
-            true
-        );
-        wp_register_script(
-            'eejs-core',
-            $this->getJsUrl(self::ASSET_NAMESPACE, 'eejs'),
-            array('ee-manifest'),
-            null,
-            true
-        );
-        wp_register_script(
-            'ee-vendor-react',
-            $this->getJsUrl(self::ASSET_NAMESPACE, 'reactVendor'),
-            array('eejs-core'),
-            null,
-            true
-        );
-        //only run this if WordPress 4.4.0 > is in use.
-        if (version_compare($wp_version, '4.4.0', '>')) {
-            //js.api
+        foreach ($this->assets->getJavascriptAssets() as $script) {
+            do_action(
+                'AHEE__EventEspresso_core_services_assets_Registry__registerScriptsAndStyles__before_script',
+                $script
+            );
             wp_register_script(
-                'eejs-api',
-                EE_LIBRARIES_URL . 'rest_api/assets/js/eejs-api.min.js',
-                array('underscore', 'eejs-core'),
-                EVENT_ESPRESSO_VERSION,
-                true
+                $script->handle(),
+                $script->source(),
+                $script->dependencies(),
+                $script->version(),
+                $script->loadInFooter()
+            );
+            if ($script->requiresTranslation()) {
+                $this->registerTranslation($script->handle());
+            }
+            if ($script->hasLocalizedData()) {
+                $this->addRegisteredScriptHandlesWithData($script->handle());
```

Comment:
This is a change of logic from the previous behaviour.  The `addRegisteredScriptHandlesWithData` prior to your changes was always done AFTER the call to `wp_localize_script` or `wp_add_inline_script`.  Here you have it happening before right?  Have you tested to make sure this doesn't change behaviour (notably you'll see duplicates of the js variable printed to the html, or missing data in the final printed data object).

Reply:
I can move this down into `enqueueData()` which also means I can remove the conditional because the call to `AssetCollection::getJavascriptAssetsWithData()` already calls `hasLocalizedData()` on each script when determining which ones to return.

I looked at how the existing code worked and didn't think it would make any difference where this was called as long as it was set somewhere.

However, I did indeed test things by visiting admin pages that utilized localized data (such as the Countries tab in the General Admin) as well as by performing some registrations on the frontend, which also utilize localized vars. I didn't notice any duplicate data, which in the past (iirc) has always resulted in the vars not working at all, but I didn't check all of the localized vars to make sure that one wasn't a duplicate. I can also do that after moving that line of code.